application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/TeXCommand.java

```diff
@@ -57,6 +59,19 @@ public void onSlashCommand(@NotNull final SlashCommandEvent event) {
         String userID = (Objects.requireNonNull(event.getMember()).getId());
         TeXFormula formula;
         try {
+            String regex = "(\\$\\S+\\$)|([a-zA-Z!., ]+)";
```

Comment:
It is linked with your comment https://github.com/Together-Java/TJ-Bot/pull/397#discussion_r812719622

There are 2 brackets which from my understanding represent 2 groups, one group matches `$...$`, which his the first bracket, the | represents or, and the second bracket ([...]+) represents the second match group. `matcher.group(2)`. 

So all together if the first match is not null, it means it has found a word with a dollar sign, then it does replace it with an empty space, while in the second match it appends `\text{ + [the words that match] + }`. 

This is all it if I am not mistaken, if I am someone please let me know.

Reply:
So first group is "i found something inside dollars" and second group is "i found something that is not a dollar"? then the second group should also be written like that `([^\\$]+)`. or similar

Reply:
Its nice that you explained it to me now, but the explanation should be obvious from the code. So that people who read the code, also understand it.

So either do some renaming, split things apart and give them variable names, ... or as last resort use comments.

Reply:
That said, I propose a different approach to the regex. While your approach is probably the most efficient, I think it is one of the less readable ways.

What you are doing now is: walk through the text, find any math-region or text-region and then do action on them.

Instead, I would just do: walk through the text, find any text-region and replace with `\text{...}`. Then, after that is fully done, simply do `.replace("$", "")`, done.

I am not convinced that you actually need to match math-regions in the first place. So your first group should be obsolete, I think.

Reply:
I changed it to this, would this be fine? I split it so it is more readable this way from this suggestion? https://github.com/Together-Java/TJ-Bot/pull/397#discussion_r813623023

```java
    private static final String MATH_REGION = "(\\$\\S+\\$)";
    private static final String TEXT_REGION = "([a-zA-Z!., ]+)";
    private static final Pattern INLINE_LATEX_REPLACEMENT = Pattern.compile(MATH_REGION + "|" + TEXT_REGION);
```

Reply:
Yes! I love that you split it up, the variable names make it very descriptive now.

I'd still add a quick comment:
```java
    // everything between 2 $'s
    private static final String MATH_REGION = "(\\$\\S+\\$)";
    // ..
    private static final String TEXT_REGION = "([a-zA-Z!., ]+)";
    private static final Pattern INLINE_LATEX_REPLACEMENT = Pattern.compile(MATH_REGION + "|" + TEXT_REGION);
    ```
    
    Just to make sure